active/0000-control-transpilation.md

```diff
@@ -0,0 +1,99 @@
+of addon code.
+
+# Motivation
+
+Ember-cli as allowed users to write ES6 code and transpile it to ES5/3 since almost the beginning
+of the project. It's one of those things that people take for granted.
+
+Today that is done through `ember-cli-babel`, an addon that is in the default blueprint and
+that the huge majority of apps use.
+
+This transpilation process has some sensitive defaults that work for most people, but also allow
+some use configuration for, by example, disable some transformations enabled by default or enable
+some experimental ones.
+
+What is less know is that this configuration only affects the application code. The transpilation
+of code living in addons is done according to the configuration of those addons. While there is
+use-cases for this, particularly addons that want to run their code though a custom babel plugin,
+generally speaking the desired transpiled output is something that the end-users should control.
+
+By example, applications that only target evergreen browsers may want to disable transpilation of
+all ES6 feature since moden browsers already ship them.
+
+This RFC wants to change this behaviour so addons by default honor the host app's preferences,
+while allowing those addons to run their own transformations.
+
+# Detailed design
+
+The idea is to still allow addons to be in control of their own transpilation process, but
+make them honour uses preferenced by default.
+
+There is three different scenarios for addons:
+
+#### The addon doesn't care about the transpilation process.
+This is the most common common situation. Most addons are authored in ES6 and they don't
+mess with the host app's configuration.
+Those addons should just transpile their code user the host's apps settings.
+
+#### Addons that require experimental features
+There is two options here. Either addons can unconditionally enable that feature only for their
+tree, or it's the user responsability to enable that feature in order to use the addon, with
+optionally the addon doing some feature detection to warn (not force) the user to enable that feature.
+
+By example, an addon that uses `async-await` which right now are only available in Chrome Canary
+could either use the host app's configuration plus this feature or it could just assume the user
+has enabled it and tell them to do so in the docs.
+
+#### Addons that add their custom babel plugins
+
+The number of addons that does this is even smaller, but one of them is nothing less than Ember Data,
+which uses a babel-plugin to strip Heimdall's instrumentation from production builds.
+
+Since addons remain in control of their transpilation and just reuse the app's configuration, addons
+can attach their own plugins.
+
+#### How can addons be in charge of their own transpilation and yet honor user preferences?
+
+Ember-cli should provide to each addon a deep clone of the configuration defined in the host app
```

Comment:
I think that the idea of deep-cloning the host's app and use that is universal, whatever version of babel we use.

There is an obvious problem if the app uses babel6 and the addon expects babel5, because the options are not compatible between them, but seems a problem of the babel upgrade more than a problem of this RFC per se.

I think that that want to customize the transpilation will have to support both babels at least for a while.